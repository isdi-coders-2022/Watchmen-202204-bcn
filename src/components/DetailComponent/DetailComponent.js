import styled from "styled-components";

const StyledDetailComponent = styled.section`
  text-align: left;
  padding: 1.2em;

  ul {
    list-style: none;
    padding-left: 0;
  }

  li {
    padding: 7px 0;
    font-weight: bolder;
  }
`;

const DetailComponent = ({ painting }) => {
  return (
    <StyledDetailComponent>
      <h2>{painting.title}</h2>
      <p>{painting.longTitle.slice(-4)}</p>
      <h3>{painting.principalOrFirstMaker}</h3>
      <p>{painting.plaqueDescriptionEnglish}</p>
      <h3>Artwork Details</h3>
      <ul>
        <li>Medium: {painting.physicalMedium}</li>
        <li>Dimensions: {painting.subTitle} </li>
        <li>Classification: {painting.objectTypes}</li>
      </ul>
    </StyledDetailComponent>
  );
};

export default DetailComponent;
