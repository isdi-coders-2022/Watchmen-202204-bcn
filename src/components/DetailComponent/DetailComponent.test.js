import { render, screen } from "@testing-library/react";
import DetailComponent from "./DetailComponent";

describe("Given a InfoComponent component", () => {
  describe("When it receives a text 'Title''", () => {
    test("Then it should render the given text", () => {
      const titleText = "Title";

      render(<DetailComponent heading1={titleText} />);
      const textElement = screen.getByRole("heading", { name: titleText });

      expect(textElement).toBeInTheDocument();
    });
  });
});
