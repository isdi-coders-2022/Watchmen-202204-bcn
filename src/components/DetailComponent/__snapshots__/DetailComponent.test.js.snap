// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Detail Component When it's rendered Then it should always match this snaptshot' 1`] = `
<section
  className="sc-bczRLJ hObVTd"
>
  <h2>
    Portrait of a Woman, Possibly Maria Trip
  </h2>
  <p>
    1639
  </p>
  <h3>
    Rembrandt van Rijn
  </h3>
  <p>
    Maria Trip, daughter of one of Amsterdam’s wealthiest merchants, was twenty when Rembrandt painted her portrait. The artist placed Maria against a stone arch and devoted particular attention to the reflected light, the fashionable dress and jewellery. The costly garments are trimmed with strips of gold lace and Maria is wearing a profusion of pearls.
  </p>
  <h3>
    Artwork Details
  </h3>
  <ul>
    <li>
      Medium: 
      oil on panel
    </li>
    <li>
      Dimensions: 
      h 107cm × w 82cm
       
    </li>
    <li>
      Classification: 
      painting
    </li>
  </ul>
</section>
`;
